bokeh/models/sources.py

```diff
@@ -136,6 +137,23 @@ def from_df(cls, data):
             new_data["index"] = index.tolist()
         return new_data
 
+    @classmethod
+    @deprecated("Bokeh 0.9.3")
```

Comment:
Well, I agree that is true in strict sense, but it's still probably more useful with the message.